src/features/vpn/services.ts

```diff
@@ -13,44 +13,37 @@ import { checkDeviceExists } from '../device-state/middleware';
 
 const { api } = sbvrUtils;
 
-const checkAuth = (() => {
-	const authQuery = _.once(() =>
-		api.resin.prepare<{ uuid: string }>({
-			resource: 'device',
-			id: {
-				uuid: { '@': 'uuid' },
-			},
-			options: {
-				$select: 'id',
-			},
-		}),
-	);
-	return multiCacheMemoizee(
-		async (uuid: string, req: permissions.PermissionReq): Promise<number> => {
-			try {
-				const device = await authQuery()({ uuid }, undefined, { req });
-				// for now if the api key is able to read the device then it has vpn access
-				return device != null ? 200 : 403;
-			} catch (err) {
-				if (err instanceof errors.UnauthorizedError) {
-					// We handle `UnauthorizedError` specially so that we can cache it as it's a relatively
-					// common case for devices that have been deleted but are still online/trying to connect
-					return 401;
-				}
-				throw err;
+const checkAuth = multiCacheMemoizee(
+	async (uuid: string, req: permissions.PermissionReq): Promise<number> => {
+		try {
+			const deviceResult = (await api.resin.post({
```

Comment:
Can't make it work w/ prepared :(
```
api.resin.prepare<{ uuid: string }>({
	url: `device(uuid=@uuid)/canAccess`,
	body: { action: 'cloudlink' },
}),
```
```
Failed to translate url:  {
	"resource": "device#canAccess",
	"key": {
		"uuid": {
			"bind": "@uuid"
		}
	},
	"property": {
		"resource": "canAccess"
	},
	"options": {}
} GET SyntaxError: Could not match resource
```

Reply:
Nevermind I was missing `method: 'POST'`...

Reply:
@thgreasi try
```ts
api.resin.prepare<{ uuid: string }>({
	method: 'POST',
	url: `device(uuid=@uuid)/canAccess`,
	body: { action: 'cloudlink' },
}),